src/main/java/chess/domain/Direction.java

```diff
@@ -0,0 +1,74 @@
+package chess.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public enum Direction {
+    UP(0, 1),
+    DOWN(0, -1),
+    RIGHT(1, 0),
+    LEFT(-1, 0),
+    UP_RIGHT(1, 1),
+    UP_LEFT(-1, 1),
+    DOWN_RIGHT(1, -1),
+    DOWN_LEFT(-1, -1),
+    UP_UP_RIGHT(1, 2),
+    UP_RIGHT_RIGHT(2, 1),
+    UP_UP_LEFT(1, -2),
+    UP_LEFT_LEFT(-2, 1),
+    DOWN_DOWN_RIGHT(1, -2),
+    DOWN_RIGHT_RIGHT(2, -1),
+    DOWN_DOWN_LEFT(-1, -2),
+    DOWN_LEFT_LEFT(-2, -1);
+    private final int column;
+    private final int row;
+
+    Direction(int column, int row) {
+        this.column = column;
+        this.row = row;
+    }
+
+    public static List<Direction> diagonalDirections() {
+        return Arrays.asList(UP_RIGHT, UP_LEFT, DOWN_RIGHT, DOWN_LEFT);
+    }
+
+    public static List<Direction> straightDirections() {
+        return Arrays.asList(UP, DOWN, LEFT, RIGHT);
+    }
+
+    public static List<Direction> aroundDirections() {
+        ArrayList<Direction> aroundDirections = new ArrayList<>(diagonalDirections());
+        aroundDirections.addAll(straightDirections());
+        return aroundDirections;
+    }
+
+    public static List<Direction> knightDirections() {
+        return Arrays.asList(
+            UP_UP_LEFT, UP_UP_RIGHT, UP_RIGHT_RIGHT, UP_LEFT_LEFT,
+            DOWN_DOWN_LEFT, DOWN_DOWN_RIGHT, DOWN_RIGHT_RIGHT, DOWN_LEFT_LEFT);
+    }
+
+    public static List<Direction> forwardDirection(TeamColor teamColor) {
+        if (teamColor == TeamColor.WHITE) {
```

Comment:
```suggestion
        if (teamColor.isWhite()) {
```
Enum에서 check용 메서드를 제공해줄 수도 있습니다 :)

Reply:
오!! 넵 알겠습니다! 감사합니다!!